prov/cxi/src/cxip_mr.c

```diff
@@ -1386,6 +1386,10 @@ static int cxip_regattr(struct fid *fid, const struct fi_mr_attr *attr,
 
 	dom = container_of(fid, struct cxip_domain, util_domain.domain_fid);
 
+	if (FI_VERSION_GE(dom->fab->util_fabric.fabric_fid.api_version, FI_VERSION(1, 22)) &&
+	    attr->page_size)
+		return -FI_ENOSYS;
+
```

Comment:
I don't think you need to update any of the providers.  If page_size is set, the provider will just ignore it (treat as 0), and the behavior will be the same as it is today.

I would keep the update to the util code, however.

Reply:
Having providers ignore page_size means page_size is a hint. Currently documentation essentially states that if page_size is set, providers must support it or -FI_ENOSYS will be returned.

I'm ok with making this a hint. Thoughts?

Reply:
Wouldn't the providers naturally support any page size already?  It may not be optimal, but it should work.

If FI_MR_ALLOCATED is set, then registrations are backed by pages, and the provider should be discovering the size.  Hopefully the discovered size matches the app value that is passed in.  This is less efficient than the provider just using the passed in value, but it should work.  Most providers fall into this category.

If FI_MR_ALLOCATED is NOT set, the situation isn't much different.  In this case, the provider is likely assuming some small page size, e.g. 4k.  That size also works for any given page size.

Reply:
Right. So page size should be a hint providers can optionally use or ignore?

Reply:
I think so.